src/plugins/intel_cpu/src/nodes/random_uniform.cpp

```diff
@@ -0,0 +1,513 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "random_uniform.hpp"
+
+#include "ie_parallel.hpp"
+#include "ie_ngraph_utils.hpp"
+#include <openvino/op/constant.hpp>
+#include <openvino/op/random_uniform.hpp>
+#include "shape_inference/custom/random_uniform.hpp"
+
+using namespace dnnl::impl::cpu;
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+
+bool RandomUniform::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        if (op->get_type_info() != op::v8::RandomUniform::get_type_info_static()) {
+            errorMessage = "Only RandomUniform operation from the opset8 is supported by the CPU plugin.";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+RandomUniform::RandomUniform(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr& context)
+        : Node(op, context, RandomUniformShapeInferFactory(op)) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        THROW_CPU_NODE_ERR(errorMessage);
+    }
+
+    // RandomUniform should generate new sequence each run even if all inputs are constants. So that method Node::IsConstant()
+    // doesn't return 'True' for RandomUniform with all constant inputs and the node generates new values for each inference,
+    // we set 'NoConst' value for 'ConstantType' in ctor.
+    constant = ConstantType::NoConst;
+
+    auto rnd_op = as_type_ptr<op::v8::RandomUniform>(op);
+    m_global_seed = rnd_op->get_global_seed();
+    m_op_seed = rnd_op->get_op_seed();
+
+    m_output_prc = op->get_output_element_type(0);
+
+    for (size_t i = 0lu; i < op->get_input_size(); i++) {
+        if (is_type<op::v0::Constant>(op->get_input_node_ptr(i))) {
+            m_const_inputs[i] = true;
+        }
+    }
+
+    if (m_const_inputs[MIN_VAL]) {
+        initEdgeValues(m_min_val, as_type<op::v0::Constant>(op->get_input_node_ptr(MIN_VAL))->get_data_ptr(), m_output_prc);
+    }
+    if (m_const_inputs[MAX_VAL]) {
+        initEdgeValues(m_max_val, as_type<op::v0::Constant>(op->get_input_node_ptr(MAX_VAL))->get_data_ptr(), m_output_prc);
+    }
+
+    if (m_algo == STL) {
+        m_generator = std::default_random_engine{static_cast<uint32_t>(m_op_seed)};
+    }
+}
+
+void RandomUniform::getSupportedDescriptors() {
+    if (getParentEdges().size() != 3) {
+        THROW_CPU_NODE_ERR("has incorrect number of input edges.");
+    }
+    if (getChildEdges().empty()) {
+        THROW_CPU_NODE_ERR("has incorrect number of output edges.");
+    }
+}
+
+void RandomUniform::initSupportedPrimitiveDescriptors() {
+    auto shape_prc = getOriginalInputPrecisionAtPort(SHAPE);
+    if (!one_of(shape_prc, InferenceEngine::Precision::I32, InferenceEngine::Precision::I64)) {
+        shape_prc = InferenceEngine::Precision::I32;
+    }
+
+    auto out_prc = getOriginalOutputPrecisionAtPort(0);
```

Comment:
As I can see kernel support bf16 only for `avx512_core_bf16` platforms. If it's so, should we add check here? Correct me please if I'm wrong

Reply:
If the kernel does not support any precision, it will not be created silently and execution will go through the CPU ref, which actually supports f16 and bf16.